dnf/repo.py

```diff
@@ -758,9 +754,35 @@ def _try_revive(self):
                     logger.debug("reviving: failed for '%s', mismatched %s sum.",
                                  self.id, algo)
                     return False
-        logger.debug("reviving: '%s' can be revived.", self.id)
+        logger.debug("reviving: '%s' can be revived - metalink checksums match.", self.id)
+        return True
+
+    def _try_revive_by_repomd(self):
+        """Use repomd to check whether our metadata are still current."""
+        repomd_fn = self.metadata.repo_dct['repomd']
+        with dnf.util.tmpdir() as tmpdir, open(repomd_fn) as repomd:
+            handle = self._handle_new_remote(tmpdir)
+            handle.yumdlist = librepo.YUM_REPOMDONLY
+            result = handle.perform()
+            fresh_repomd_fn = result.rpmmd_repo['repomd']
+            with open(fresh_repomd_fn) as fresh_repomd:
+                if repomd.read() != fresh_repomd.read():
+                    logger.debug("reviving: failed for '%s', mismatched repomd.",
+                             self.id)
+                    return False
+        logger.debug("reviving: '%s' can be revived - repomd matches.", self.id)
         return True
 
+    def _try_revive(self):
+        """Use metalink to check whether our metadata are still current."""
+        if not self.metadata:
```

Comment:
No? It's ok if self.metalink is None, but if metadata is None, then there is nothing for us to verify in the first place :).